### PR TITLE
APIM-1625 cache policy v4

### PR DESCRIPTION
**Pull request by leleueri**
Fix APIM-1625

**Description**

Migrate the cache policy to use v4 interfaces.

---
**Gravitee.io Automatic Deployment**

🚀 A prerelease version of this package has been published on Gravitee's private artifactory, you can:
 - use it directly by updating your project with version: `2.0.0-apim-1625-cache-policy-v4-SNAPSHOT`
 - download it from Artifactory [here](https://odbxikk7vo-artifactory.services.clever-cloud.com/gravitee-snapshots/io/gravitee/policy/gravitee-policy-cache/2.0.0-apim-1625-cache-policy-v4-SNAPSHOT/gravitee-policy-cache-2.0.0-apim-1625-cache-policy-v4-SNAPSHOT.zip)

### CODE REVIEW

**Review by jhaeyaert — CHANGES_REQUESTED** [review]
I've put some comments. 

More of a global reflection because I think it is out of the scope of this PR but I think we should provide a cache abstraction that works natively with RxJava to avoid switching the scheduler. Doing that kind of switch isn't optimized when the underlying cache implementation is an in-memory cache. This should be the responsibility of the cache implementation to decide whether it is necessary or not.

File: src/main/java/io/gravitee/policy/cache/invoker/CacheInvoker.java [hunk 1]
```diff
@@ -0,0 +1,280 @@
...
+                    resp.setStatus(response.status());
+                    resp.setHeaders(httpHeaders);
+
+                    long timeToLive = -1;
+                    if (cachePolicyConfiguration.isUseResponseCacheHeaders()) {
+                        timeToLive = resolveTimeToLive(response);
+                    }
+                    if (timeToLive == -1 || cachePolicyConfiguration.getTimeToLiveSeconds() < timeToLive) {
+                        timeToLive = cachePolicyConfiguration.getTimeToLiveSeconds();
+                    }
```
[the remark this answers is not in the document]
Comment (leleueri) [reply src/main/java/io/gravitee/policy/cache/invoker/CacheInvoker.java:185]: indeed

File: src/main/java/io/gravitee/policy/cache/invoker/CacheInvoker.java [hunk 2]
```diff
@@ -0,0 +1,280 @@
...
+    @Override
+    public String getId() {
+        return CACHE_ENDPOINT_INVOKER_ID;
+    }
+
+    @Override
+    public Completable invoke(ExecutionContext executionContext) {
+        // Here we have to check if there is a value in cache
+        var cacheId = hash(executionContext);
+        LOGGER.debug("Looking for element in cache with the key {}", cacheId);
```
[the remark this answers is not in the document]
Comment (leleueri) [reply src/main/java/io/gravitee/policy/cache/invoker/CacheInvoker.java:93]: I do prefer to avoid lombok. just to avoid declaring a logger.

**Review by jhaeyaert — APPROVED** [review]

**Pull request merged**